src/request_collector/server.py

```diff
@@ -24,26 +43,54 @@ def error_handler(context, exc_info):
     sys.exit()
 
 
+def setup_matrix(private_key: str) -> Tuple[GMatrixClient, Room]:
+    available_servers_url = DEFAULT_MATRIX_KNOWN_SERVERS[Environment.DEVELOPMENT]
+    available_servers = get_matrix_servers(available_servers_url)
+
+    def _http_retry_delay() -> Iterable[float]:
+        # below constants are defined in raiden.app.App.DEFAULT_CONFIG
+        return udp_utils.timeout_exponential_backoff(
+            DEFAULT_TRANSPORT_RETRIES_BEFORE_BACKOFF,
+            DEFAULT_TRANSPORT_MATRIX_RETRY_INTERVAL / 5,
+            DEFAULT_TRANSPORT_MATRIX_RETRY_INTERVAL,
+        )
+
+    client = make_client(
+        servers=available_servers,
+        http_pool_maxsize=4,
+        http_retry_timeout=40,
+        http_retry_delay=_http_retry_delay,
+    )
+    login_or_register(client, signer=LocalSigner(private_key=decode_hex(private_key)))
+
+    monitoring_room = join_global_room(
+        client=client,
+        name=MONITORING_BROADCASTING_ROOM,
+        servers=available_servers,
+    )
+
+    return client, monitoring_room
+
+
 class RequestCollector(gevent.Greenlet):
     def __init__(
         self,
+        private_key: str,
         state_db: SharedDatabase,
-        transport: Transport,
     ):
         super().__init__()
 
-        assert isinstance(transport, Transport)
-
+        self.private_key = private_key
         self.state_db = state_db
-        self.transport = transport
+
         self.stop_event = gevent.event.Event()
-        self.transport.add_message_callback(lambda message: self.on_message_event(message))
 
         self.task_list: List[gevent.Greenlet] = []
+        self.client, self.monitoring_room = setup_matrix(self.private_key)
+        self.monitoring_room.add_listener(self._handle_message, 'm.room.message')
 
     def _run(self):
         register_error_handler(error_handler)
-        self.transport.start()
 
         # this loop will wait until spawned greenlets complete
         while self.stop_event.is_set() is False:
```

Comment:
Instead of `gevent.sleep(1)` inside the while block with the `is_set` verification on the condition, I'd suggest `while self.stop_event.wait(1) is not True`, which will return immediatelly with `True` iff the event was set, instead of blocking up to 1sec if the event was set in the middle of a loop

Reply:
Wouldn't that change the idea here? Currently this only sleeps when no tasks are waiting, otherwise it waits for the next task to complete.
With your change (and my understanding, which might be wrong) this will always wait in the loop, even if there are tasks to be completed.

Reply:
Oh, right, that would, didn't notice you wanted to loop also through existing tasks. But anyway you could use `stop_event.wait` as a _gevent.sleep_ which returns on also on set and break when True

Reply:
So you say further down I'd rather write:
```py
            if len(tasks) == 0:
                self.stop_event.wait(1)
                continue
```